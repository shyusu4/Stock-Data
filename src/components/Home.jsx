import Header from './Header';
import StockDisplay from './StockDisplay';
import homepageimg from '../img/home.jpg'

function Home() {
  return (
    <div>
      <Header />
      <div className='img-container'>
      <img className='homepageimg' src={homepageimg} alt='stocks'></img>
      <div className="main-text">The real earning starts here.</div>
      <p className='caption'>find real-time stock prices below <span>&#8595;</span></p>
      </div>
      <div className='stats'>Stats by company</div>
      <StockDisplay />
    </div>
  );
}

export default Home;