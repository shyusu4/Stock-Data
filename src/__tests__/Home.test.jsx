import React from 'react';
import Renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Home from '../components/Home';

// eslint-disable-next-line func-names
window.matchMedia = window.matchMedia || function () {
  return {
    matches: false,
    addListener() {},
    removeListener() {},
  };
};

describe('Home', () => {
  it('Check if Home component renders correctly', () => {
    const tree = Renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Home />
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
