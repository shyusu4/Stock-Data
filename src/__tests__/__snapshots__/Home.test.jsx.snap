// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Check if Home component renders correctly 1`] = `
<div>
  <nav
    className="mb-3 fixed-top navbar navbar navbar-expand-lg navbar-dark"
  >
    <div
      className="container-fluid"
    >
      <a
        className="navbar-name navbar-brand"
        href="#"
      >
        Stock Market Metrics
      </a>
      <button
        aria-controls="offcanvasNavbar-expand-lg"
        aria-label="Toggle navigation"
        className="navbar-toggler collapsed"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        aria-labelledby="offcanvasNavbarLabel-expand-lg"
        className="offcanvas offcanvas-end"
        id="offcanvasNavbar-expand-lg"
      >
        <div
          className="offcanvas-header"
        >
          <div
            className="offcanvas-title h5"
            id="offcanvasNavbarLabel-expand-lg"
          >
            Stock Market Metrics
          </div>
          <button
            aria-label="Close"
            className="btn-close"
            onClick={[Function]}
            type="button"
          />
        </div>
        <div
          className="offcanvas-body"
        >
          <div
            className="justify-content-end flex-grow-1 pe-3 navbar-nav"
            onKeyDown={[Function]}
          >
            <a
              className="nav-link"
              data-rr-ui-event-key="#"
              href="#"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={0}
            >
              Home
            </a>
          </div>
        </div>
      </div>
    </div>
  </nav>
  <div
    className="img-container"
  >
    <img
      alt="stocks"
      className="homepageimg"
      src="home.jpg"
    />
    <div
      className="main-text"
    >
      The real earning starts here.
    </div>
    <p
      className="caption"
    >
      find real-time stock prices below
       
      <span>
        â†“
      </span>
    </p>
  </div>
  <div
    className="stats"
  >
    <form
      className="d-flex"
    >
      <p
        className="stats-text"
      >
        Stats by company
      </p>
      <input
        aria-label="Search"
        className="searchbar form-control"
        onChange={[Function]}
        placeholder="Search by company name or symbol"
        type="search"
      />
    </form>
  </div>
  <div
    className="stock-container"
  />
</div>
`;
